src/resource_sqlite.h

```diff
@@ -33,16 +33,92 @@
 
 #include "core/io/resource.h"
 #include "core/variant/variant.h"
+#include "godot_sqlite.h"
+#include "core/variant/typed_array.h"
+
+class SQLiteColumnSchema: public RefCounted {
+    GDCLASS(SQLiteColumnSchema, RefCounted);
+    String name;
+    Variant::Type type;
+    Variant default_value;
+    bool primary_key = false;
+    bool auto_increment = false;
+    bool not_null = false;
+    bool unique = false;
+
+protected:
+    static void _bind_methods() {
+        ClassDB::bind_method(D_METHOD("get_name"), &SQLiteColumnSchema::get_name);
+        ClassDB::bind_method(D_METHOD("get_type"), &SQLiteColumnSchema::get_type);
+        ClassDB::bind_method(D_METHOD("get_default_value"), &SQLiteColumnSchema::get_default_value);
+        ClassDB::bind_method(D_METHOD("is_primary_key"), &SQLiteColumnSchema::is_primary_key);
+        ClassDB::bind_method(D_METHOD("is_auto_increment"), &SQLiteColumnSchema::is_auto_increment);
+        ClassDB::bind_method(D_METHOD("is_not_null"), &SQLiteColumnSchema::is_not_null);
+        ClassDB::bind_method(D_METHOD("is_unique"), &SQLiteColumnSchema::is_unique);
+
+        ClassDB::bind_method(D_METHOD("set_name", "name"), &SQLiteColumnSchema::set_name);
+        ClassDB::bind_method(D_METHOD("set_type", "type"), &SQLiteColumnSchema::set_type);
+        ClassDB::bind_method(D_METHOD("set_default_value", "default_value"), &SQLiteColumnSchema::set_default_value);
+        ClassDB::bind_method(D_METHOD("set_primary_key", "primary_key"), &SQLiteColumnSchema::set_primary_key);
+        ClassDB::bind_method(D_METHOD("set_auto_increment", "auto_increment"), &SQLiteColumnSchema::set_auto_increment);
+        ClassDB::bind_method(D_METHOD("set_not_null", "not_null"), &SQLiteColumnSchema::set_not_null);
+        ClassDB::bind_method(D_METHOD("set_unique", "unique"), &SQLiteColumnSchema::set_unique);
+
+        ADD_PROPERTY(PropertyInfo(Variant::STRING, "name"), "set_name", "get_name");
+        ADD_PROPERTY(PropertyInfo(Variant::INT, "type"), "set_type", "get_type");
+        ADD_PROPERTY(PropertyInfo(Variant::STRING, "default_value"), "set_default_value", "get_default_value");
+        ADD_PROPERTY(PropertyInfo(Variant::BOOL, "primary_key"), "set_primary_key", "is_primary_key");
+        ADD_PROPERTY(PropertyInfo(Variant::BOOL, "auto_increment"), "set_auto_increment", "is_auto_increment");
+        ADD_PROPERTY(PropertyInfo(Variant::BOOL, "not_null"), "set_not_null", "is_not_null");
+        ADD_PROPERTY(PropertyInfo(Variant::BOOL, "unique"), "set_unique", "is_unique");
+
+        ClassDB::bind_static_method("SQLiteColumnSchema", D_METHOD("create", "name", "type", "default_value", "primary_key", "auto_increment", "not_null", "unique"), &SQLiteColumnSchema::create, DEFVAL(Variant::Type::STRING), DEFVAL(Variant()), DEFVAL(false), DEFVAL(false), DEFVAL(false), DEFVAL(false));
```

Comment:
See also https://www.sqlite.org/stricttables.html

Reply:
Can be extended in the future I hope. For now I only added a subset.